src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -99,7 +99,7 @@ protected override void OnPersist(ApplicationEngine engine)
         {
             base.OnPersist(engine);
             StorageItem storage = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_NextValidators), () => new StorageItem());
-            storage.Value = GetValidators(engine.Snapshot).ToByteArray();
+            storage.Set(GetValidators(engine.Snapshot));
```

Comment:
I think it'd be nice to optimize this one also while we're here.

If you're to look at preview2 state change dump, you'll see something like
```
 {
  "block": 3,
  "size": 1,
  "storage": [
   {
    "state": "Changed",
    "key": "ffffffff0e00000000000000000000000000000001",
    "value": "e807023e9b32ea89b94d066e649b124fd50e396ee91369e8e2a6ae1b11c170d022256d03009b7540e10f2562e5fd8fac9eaec25166a58b26e412348ff5a86927bfac22a202ba2c70f5996f357a43198705859fae2cfea13e1172962800772b3d588a9d4abd03408dcd416396f64783ac587ea1e1593c57d9fea880c8a6a1920e92a25947780602a7834be9b32e2981d157cb5bbd3acb42cfd11ea5c3b10224d7a44e98c5910f1b0214baf0ceea3a66f17e7e1e839ea25fd8bed6cd82e6bb6e68250189065f44ff01030205e9cefaea5a1dfc580af20c8d5aa2468bb0148f1a5e4605fc622c80e604ba00"
   }
  ]
 },
```
for every block and this value obviously doesn't change often. I guess it would be same even with an updated (post-preview2) governance system. But we shouldn't update storage at all if this value doesn't change, it's not hard to do, actually the only reason we didn't do it immediately in neo-go is that it would break state change dumps compatibility for us (even though technically storage contents would be the same for each block).

Reply:
The problem is that TrackeableState is outside StorageItem, I know what you say, I will try to improve it in other PR.